app/jobs/create_virtual_objects_job.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+##
+# job to create virtual objects
+class CreateVirtualObjectsJob < GenericJob
+  queue_as :default
+
+  ##
+  # A job that creates virtual objects
+  # @param [Integer] bulk_action_id GlobalID for a BulkAction object
+  # @param [Hash] params additional parameters that an Argo job may need
+  # @option params [String] :create_virtual_objects CSV string
+  def perform(bulk_action_id, params)
+    super
+
+    virtual_objects = CsvToJsonService.convert(csv_string: params[:create_virtual_objects])
+
+    with_bulk_action_log do |log|
+      log.puts("#{Time.current} Starting #{self.class} for BulkAction #{bulk_action_id}")
+
+      parent_druids = virtual_objects[:virtual_objects].map { |hash| hash[:parent_id] }
+
+      # NOTE: We use this instead of `update_druid_count` because virtual object
+      #       creation does not use the `pids` form field.
+      bulk_action.update(druid_count_total: parent_druids.length)
+
+      # rubocop:disable Metrics/LineLength
+      problematic_druids = find_and_authorize_druids(parent_druids)
+      if problematic_druids[:unauthorized].any? || problematic_druids[:not_found].any?
+        log.puts("#{Time.current} Could not create virtual objects because user lacks ability to manage the following parent druids: #{problematic_druids[:unauthorized].to_sentence}") if problematic_druids[:unauthorized].any?
+        log.puts("#{Time.current} Could not create virtual objects because the following parent druids were not found: #{problematic_druids[:not_found].to_sentence}") if problematic_druids[:not_found].any?
+        bulk_action.update(druid_count_fail: parent_druids.length)
+        return
```

Comment:
so the code is basically only going to run if ALL the parent druids are ok.   Which is fine as a first pass but I'm wondering if Andrew would prefer that the ok parent druids get run (no idea - I'm ok w whatever he wants.  I can see the value in "we'll either run this csv, or we won't.  do your homework."

Reply:
I'm changing this behavior so it runs all non-problematic druids.